code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -62,50 +62,51 @@
 		return
 	if(folded)
 		return
-
 	if(user.do_actions)
 		return
+	if(!isturf(target))
+		return
+
+	var/turf/target_turf = target
+
+	if(dirt_amt)
+		balloon_alert(user, "Dumps the [dirt_type == DIRT_TYPE_SNOW ? "snow" : "dirt"]")
+		playsound(user.loc, "rustle", 30, 1, 6)
+		if(dirt_type == DIRT_TYPE_SNOW)
+			var/obj/item/stack/snow/S = locate() in target_turf
+			if(S && (S.amount + dirt_amt < S.max_amount))
+				S.amount += dirt_amt
+			else
+				new /obj/item/stack/snow(target_turf, dirt_amt)
+		dirt_amt = 0
+		update_icon()
```

Comment:
we use update appearance now

Reply:
Only if we need to update other stuff? We call update_icon all over the place still